### PR TITLE
 Updates what kaggle_secrets expects to receive from Web Tier, and adds timeout.

### PR DESCRIPTION
**Pull request by vimota**

### CODE REVIEW

File: patches/kaggle_secrets.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
 _KAGGLE_DEFAULT_URL_BASE = "https://www.kaggle.com"
 _KAGGLE_URL_BASE_ENV_VAR_NAME = "KAGGLE_URL_BASE"
 _KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME = "KAGGLE_USER_SECRETS_TOKEN"
+TIMEOUT_SECS = 10
```
Comment (ifigotin) [inline patches/kaggle_secrets.py]: Let's increase this one to say 30-40? I think 10s may not be enough.

File: patches/kaggle_secrets.py [hunk 2]
```diff
@@ -43,9 +44,12 @@ def _make_post_request(self, data):
         req = urllib.request.Request(url, headers=self.headers, data=bytes(
             json.dumps(request_body), encoding="utf-8"))
         try:
-            with urllib.request.urlopen(req) as response:
+            with urllib.request.urlopen(req, timeout=TIMEOUT_SECS) as response:
                 response_json = json.loads(response.read())
-                return response_json
+                if not response_json.get('wasSuccessful') or 'result' not in response_json:
```
Comment (ifigotin) [inline patches/kaggle_secrets.py]: Just a sanity check: is `response_json.get('wasSuccessful')` going to return False, if the string value in the JSON is 'false'? 
I'd like to make sure that the string "false" gets interpreted as False. 
(e.g. https://stackoverflow.com/questions/715417/converting-from-a-string-to-boolean-in-python)

File: tests/test_user_secrets.py [hunk 3]
```diff
@@ -49,7 +49,7 @@ def set_request(self):
                 _request['headers'] = self.headers
 
             def get_response(self):
-                return {"Secret": secret}
+                return {'result': {'secret': secret, 'secretType': 'refreshToken', 'secretProvider': 'google'}, 'wasSuccessful': True}
```
Comment (ifigotin) [inline tests/test_user_secrets.py]: Could you add a test with 'wasSuccessfull' : 'false' as well?

File: patches/kaggle_secrets.py (hunk 1 above)
Comment (vimota) [reply patches/kaggle_secrets.py]: Done.

File: patches/kaggle_secrets.py (hunk 2 above)
Comment (vimota) [reply patches/kaggle_secrets.py]: Done. Added test case.

File: tests/test_user_secrets.py (hunk 3 above)
Comment (vimota) [reply tests/test_user_secrets.py]: Done.

**Pull request merged**